app/controllers/users.controller.js

```diff
@@ -284,3 +284,13 @@ exports.isUserBanned = function (req, res, next) {
         next();
     });
 };
+
+exports.getCurrentUser = function (req, res) {
+    var user = AuthUtils.getUserFromToken(req);
+    User.findById(user._id, function (err, user) {
+        if (err || !user) {
+            return res.status(500).send('Failed to find current user.');
```

Comment:
In the case of `!user`, the status code should be a 400. 500 is fine for `err`, though. A 500 status code is for when something goes wrong internally with the back end, a 400 is when the `req` has bad info in it.

I realize one of the other methods does it this same way. I missed it when doing code review on one of Scott's PRs

Reply:
Something like:

`
if (err) {
    return res.status(500).send(err.message);
} else if (!user) {
    return res.status(400).send('Failed to find current user.');
}
`